### ISSUE TITLE
Expression parsing in offsets when dealing with constants doesn't work

### ISSUE THREAD

**Issue by kelloggm** [kelloggm/checker-framework#120]
Consider this code:

```java
    void offset_test2() {
	int[] a = new int[4];
	int k = 5;
	@LTLengthOf(value="a", offset="-k") int i = 6;
    }
```

`i` does in fact have the correct type here, but the common assignment checks don't actually do any parsing of the expression in the offset, so the assignment is rejected. If `k` in the offset is replaced with `5`, the assignment succeeds.

**Comment by kelloggm** [comment]
I'm not sure how serious this problem is but I noticed it while writing tests for #109

**Comment by smillst** [comment]
This should only be a problem for explicitly written offsets.  If the offset is inferred, both the constant value and the expression appear as offsets.  So I'm going to mark this as low, but if it comes up in test cases, I can fix it.

**Comment by kelloggm** [comment]
I agree with that course of action. I just wanted to document that I'd
encountered it.
